lib/history-index.js

```diff
@@ -0,0 +1,110 @@
+var duplexify = require('duplexify')
+var through = require('through2')
+
+function createIndex (osm, db) {
+  return {
+    maxBatch: 100,
+    map: function (nodes, next) {
+      var batch = []
+      nodes.forEach(function (node) {
+        if (node.value.type !== 'osm/element') return
+        if (!node.value.element) return
+        var key = node.value.element.type === 'observation'
+          ? 'created_at' : 'timestamp'
+        var timestamp = node.value.element[key]
+        if (!timestamp) return
+        // dates should be stored as ISO format which lexicographically sort
+        // but we'll convert and re-encode just to be extra sure
+        var date = new Date(timestamp)
+        try {
+          var iso = date.toISOString()
+        } catch (err) {
+          return
+        }
+        batch.push({
+          type: 'put',
+          key: 'a!' + iso,
+          value: node.key + '@' + node.seq
+        })
+        var t = String(node.value.element.type || '').replace(/!/g, '_')
```

Comment:
Is there a benefit to indexing malformed nodes that lack a type?

Reply:
I can't think of one. I added that so it wouldn't crash if a node didn't have a type, but I've changed it so it won't write the record in the first place too.